### PR TITLE
Flatcar support

### PR DESCRIPTION
**Pull request by mazzy89**
Recognize Flatcar (https://www.flatcar-linux.org/)

Flatcar is a friendly-fork of CoreOS. 

It is very similar to CoreOS in terms of configuring it. However, I've decided anyway to treat like a different distribution and create different if-else branches.

### CODE REVIEW

**Comment by mazzy89** [comment]
I've signed it

File: nodeup/pkg/model/kubelet.go [hunk 1]
```diff
@@ -336,6 +343,30 @@ func (b *KubeletBuilder) addStaticUtils(c *fi.ModelBuilderContext) error {
 		}
 	}
 
+	if b.Distribution == distros.DistributionFlatcar {
+		// Flatcar does not ship with conntrack.  Install our own (statically linked) version
+		// TODO: Extract to common function?
+		for _, binary := range []string{"conntrack"} {
```
Comment (mazzy89) [inline nodeup/pkg/model/kubelet.go]: I've checked and I've found out that Flatcar is shipped with `socat` so it is not necessary to install it.

**Comment by mazzy89** [comment]
Thank you @chrisz100. Ready to be merged.

**Comment by rifelpet** [comment]
Can we include a section in the documentation for Flatcar? 

https://github.com/kubernetes/kops/blob/master/docs/images.md

**Comment by mazzy89** [comment]
Yes sure. I will

**Comment by mazzy89** [comment]
Added a section in the doc

File: nodeup/pkg/distros/distribution.go [hunk 2]
```diff
@@ -52,6 +53,8 @@ func (d Distribution) BuildTags() []string {
 		t = []string{"_rhel7"}
 	case DistributionCoreOS:
 		t = []string{"_coreos"}
+	case DistributionFlatcar:
+		t = []string{"_flatcar"}
```
Comment (justinsb) [inline nodeup/pkg/distros/distribution.go]: Just FYI we hopefully aren't using these tags any more.  But no problem with leaving this in for now and removing once we confirm we're not using tags :-)

File: nodeup/pkg/model/kubelet.go (hunk 1 above)
Comment (justinsb) [reply nodeup/pkg/model/kubelet.go]: That is really good news - the omission of socat was so surprising!

**Comment by justinsb** [comment]
Thanks @mazzy89 - looks good.  I do wonder if we should fold some of the very similar code, but let's run with it and see how it evolves!

/approve
/lgtm

**Pull request merged**